src/pybind/mgr/dashboard/frontend/e2e/rgw/buckets.po.ts

```diff
@@ -1,5 +1,183 @@
+    browser
+      .actions()
+      .mouseMove(editButton)
+      .click()
+      .perform()
+      .then(() => {
+        this.navigateTo();
+        browser
+          .wait(Helper.EC.elementToBeClickable(PageHelper.getTableCell(name)), 10000)
+          .then(() => {
+            PageHelper.getTableCell(name).click();
+            expect(
+              element
+                .all(by.css('.table.table-striped.table-bordered'))
+                .first()
+                .getText()
+            ).toMatch(new_owner);
+          }); // click edit button and wait to make sure new owner is present in table
+      });
+  }
+
+  static delete(name) {
+    this.navigateTo();
+
+    // wait for table to load
+    browser.wait(Helper.EC.elementToBeClickable(PageHelper.getTableCell(name)), 10000);
+
+    PageHelper.getTableCell(name).click(); // click on the bucket you want to delete in the table
+    $('.table-actions button.dropdown-toggle').click(); // click toggle menu
+    $('li.delete a').click(); // click delete
+    // wait for pop-up to be visible (checks for title of pop-up)
+    browser
+      .wait(Helper.EC.visibilityOf(element(by.css('.modal-title.pull-left'))), 10000)
+      .then(() => {
+        $('#confirmation').click(); // click confirmation checkbox
+        element(by.cssContainingText('button', 'Delete bucket'))
+          .click()
+          .then(() => {
+            this.navigateTo();
+            browser.wait(Helper.EC.not(Helper.EC.presenceOf(PageHelper.getTableCell(name))), 10000);
+          });
+      });
+  }
+
+  static invalidCreate() {
+   // Types in an invalid name for a bucket
+   this.navigateTo('create');
+   expect(PageHelper.getTitleText()).toBe('Create Bucket');
+   browser.driver
+     .manage()
+     .window()
+     .maximize();
+   const nameBox = element(by.id('bid')); // Grabs name box field
+
+   // Clears input field and then puts an invalid input, then has a wait function
+   // to verify that the error pops up
+   nameBox.clear().then(() => {
+     nameBox.sendKeys('rq').then(() => {
+       browser
+         .wait(function() { // Waiting for website to decide if name is valid or not
+           return nameBox.getAttribute('class').then(function(classValue) {
+             return classValue.indexOf('ng-pending') === -1;
+           });
+         }, 6000)
+         .then(() => {
+           expect(nameBox.getAttribute('class')).toContain('ng-invalid');
+         });
+     });
+   });
+
+   browser.driver
+     .manage()
+     .window()
+     .maximize();
+   // Chooses 'Select a user' rather than a valid owner on Create Bucket page
+   // and checks if it's an invalid input
+   const ownerbox = element(by.id('owner'));
+   browser.wait(Helper.EC.elementToBeClickable(ownerbox), 5000);
+   browser.actions().mouseMove(ownerbox).click(); // Clicks the Owner drop down on the Create Bucket page
+   ownerbox.all(by.css('option')).first().click(); // select the first option, which is invalid
```

Comment:
```suggestion
   ownerbox.all(by.css('option')).first().click(); // select the first option, which is invalid because the entry is a placeholder
```